### PR TITLE
Use redux in react components

### PR DESCRIPTION
**Pull request by ahzamir**
### In this pull request I have made some changes to use redux in my react components according to the requirements:

- I have made my state immutable in reducers!
- I have changed the default state in my books reducer from an empty array to an array with a few books.
- I have Imported Redux Provider and my store into the main component of my React app.
- I have connected my app to the Redux store.
- The list of books displays books from the Redux store successfully.
  - I have used useSelector to consume the state.
- In my React component responsible for adding new books, I have added functionality for "Add new" button click:
and Imported Redux useDispatch.
  - I have set the data inputs in the local React state (set title and author).
  - I have dispatched a corresponding action for add book action.
- In my React component responsible for removing books, I have added functionality for "Remove" button click:
  - I have Imported Redux useDispatch.
  - I have dispatched a corresponding action for removing books.
- In my React component responsible for categories, I have added functionality for "Check status" button click:
  - I have Imported Redux useDispatch.
  - I have dispatched a corresponding action for returning the status.

### CODE REVIEW

File: src/components/pages/CategoriesPage.js [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
+  const dispatch = useDispatch();
+  return (
+    <button
+      type="button"
+      onClick={() => dispatch(checkStatus())}
+    >
+      Check status
+    </button>
+  );
+};
```
Comment (AmaduKamara) [inline src/components/pages/CategoriesPage.js]: - [ ] Your implementation of the check status is almost done but there is one more work to be done. Kindly display the "User construction" message when the Check status button is clicked. It is supposed to fetch that message from the store and display it on the category page when the button is clicked. Please work on that and your project will be in a complete state 👍

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️♻️♻️ 

Hi @ahzamir,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ Correct use of Gitflow
✅ No linter error
✅ Good commit messages
✅ Good descriptive PR

## Required Changes ♻️
_Check the comments under the review._

## Suggested Changes ♻️
_Check the comments under the review._

- [ ] Kindly edit your README to have content relating to your project. Currently, you are using the right template but it just has dummy template texts. You are to use this template format and fill in every project detail you working on. 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @ahzamir, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
- [ ] Kindly edit the info of the README file as the previous reviewer asked `Project Name`, `Description`, `Build with` and `Live Demo` if exist.
image



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Zamir, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**